test/test_fake_tensor.py

```diff
@@ -165,6 +165,21 @@ def test_index_cuda_with_cpu(self):
             out = x[torch.zeros([36], dtype=torch.int64)]
             self.checkType(out, "cuda", [36])
 
+    @unittest.skipIf(
+        TEST_WITH_TORCHDYNAMO, "isinstance check for FakeTensor won't work with compile"
+    )
+    @unittest.skipIf(not RUN_CUDA, "requires cuda")
+    @parametrize(
+        "fp8_dtype",
+        [torch.float8_e4m3fn, torch.float8_e5m2],
+    )
+    def test_index_cuda_fp8(self, fp8_dtype):
```

Comment:
can we combine with an existing test instead?
* Option 1: extend `test_basic` to also use cuda and add float8 dtypes
* Option 2: extend `test_index_cuda_with_cpu` with float8 dtypes

Having one-off tests for specific dtypes is good to avoid if possible.